serde-oracle-jdbc-json/src/main/java/io/micronaut/serde/oracle/jdbc/json/OracleJdbcJsonParserDecoder.java

```diff
@@ -167,11 +163,127 @@ protected void skipChildren() {
     }
 
     @Override
-    public IOException createDeserializationException(String message, Object invalidValue) {
+    @NonNull
+    public IOException createDeserializationException(@NonNull String message, @Nullable Object invalidValue) {
         if (invalidValue != null) {
             return new InvalidFormatException(message, null, invalidValue);
         } else {
             return new SerdeException(message);
         }
     }
+
+    /**
+     * Decodes Oracle JSON binary data as byte array.
+     *
+     * @return the byte array for Oracle JSON binary
+     */
+    public byte[] decodeBinary() {
+        if (currentEvent == OracleJsonParser.Event.VALUE_BINARY) {
+            byte[] bytes = jsonParser.getBytes();
+            nextToken();
+            return bytes;
+        }
+        if (currentEvent == OracleJsonParser.Event.START_ARRAY) {
+            OracleJsonArray oracleJsonArray = jsonParser.getArray();
+            int size = oracleJsonArray.size();
+            byte[] bytes = new byte[size];
+            for (int i = 0; i < size; i++) {
+                if (oracleJsonArray.isNull(i)) {
+                    bytes[i] = 0;
+                } else {
+                    bytes[i] = (byte) oracleJsonArray.getInt(i);
+                }
+            }
+            nextToken();
+            return bytes;
+        }
+        if (currentEvent == OracleJsonParser.Event.VALUE_STRING) {
+            String str = jsonParser.getString();
```

Comment:
I wanted to avoid this, but for tests that parse serialized JSON with byte[] we need to have this because byte[] needs to be base16 encoded when writing to the db so for the test for all decoders we need to be able to decode it back and verify.